Yank/multistate/multistateanalyzer.py

```diff
@@ -2009,7 +2034,7 @@ def _decorrelated_N_l(self, instance):
     _unbiased_decorrelated_u_ln = CachedProperty(
         name='unbiased_decorrelated_u_ln',
         dependencies=['unbias_restraint', 'restraint_energy_cutoff', 'restraint_distance_cutoff',
-                      'decorrelated_state_indices_ln', 'decorrelated_u_ln', 'decorrelated_N_l'],
+                      'decorrelated_state_indices_ln', 'decorrelated_u_ln', 'decorrelated_N_l', 'use_full_trajectory'],
```

Comment:
I wouldn't add a dependency on `use_full_trajectory` here, in `_unbiased_decorrelated_N_l`, and `mbar`. Only the properties that use `use_full_trajectory` explicitly in the function should. The chain of dependency will be correctly resolved.

`decorrelated_state_indices_ln` should depend on `use_full_trajectory` though through `self._decorrelated_iterations`.